GDJS/Runtime/CustomRuntimeObject.ts

```diff
@@ -27,11 +27,15 @@ namespace gdjs {
     _untransformedHitBoxes: gdjs.Polygon[] = [];
     /** The dimension of this object is calculated from its children AABBs. */
     _unrotatedAABB: AABB = { min: [0, 0], max: [0, 0] };
-    _scaleX: number = 1;
-    _scaleY: number = 1;
+    _scaleX: float = 1;
+    _scaleY: float = 1;
     _flippedX: boolean = false;
     _flippedY: boolean = false;
     opacity: float = 255;
+    _customCenter: FloatPoint | null = null;
+    _localTransformation: gdjs.AffineTransformation = new gdjs.AffineTransformation();
+    _localInverseTransformation: gdjs.AffineTransformation = new gdjs.AffineTransformation();
+    _isLocalTransformationDirty: boolean = true;
```

Comment:
It will take more memory but the code is easier to maintain and maybe faster because an affine transformation is 8 arithmetical operations versus something like 25 for the hand crafted transformation.

Reply:
I think that's fine and fair to have these. Custom objects are not designed to be super lightweight objects working as particles for example.